README.md

```diff
@@ -17,75 +17,81 @@ This is a general purpose library for working with Content Security Policy polic
+
+
 ### Create a Policy
 
-Parse a policy using one of the `Parser.parse` static methods. An `Origin` or `String` may be given as the origin. The third parameter, if given, will be populated with notices.
+Parse a policy using either `Policy.parseSerializedCSP` or `Policy.parseSerializedCSPList`. The second parameter will be called for each warning or error.
 
 ```java
-ArrayList<Notice> notices = new ArrayList<>();
-Origin origin = URI.parse("http://example.com");
-String policyText = "...";
-Policy p = Parser.parse(policyText, origin, notices);
+String policyText = "script-src 'none'";
+Policy policy = Policy.parseSerializedCSP("default-src 'none'", (severity, message, directiveIndex, valueIndex) -> {
+  System.err.println(severity.name() + " at directive " + directiveIndex + (valueIndex == -1 ? "" : " at value " + valueIndex) + ": " + message);
+});
 ```
 
-To include location information, use `ParserWithLocation.parse` in place of `Parser.parse`.
+### Query a Policy
+
+The high-level querying methods allow you to specify whatever relevant information you have. The missing information will be assumed to be worst-case - that is, these methods will return `true` only if any object which matches the provided characteristics would be allowed, regardless of its other characteristics. 
 
 ```java
-ArrayList<Notice> notices = new ArrayList<>();
-ParserWithLocation.parse("image-src *; script-src none; report-uri /report", "https://example.com", notices);
+Policy policy = Policy.parseSerializedCSP("script-src http://a", Policy.PolicyErrorConsumer.ignored);
+
+// true
+System.out.println(policy.allowsExternalScript(
+  Optional.empty(),
+  Optional.empty(),
+  Optional.of(URI.parse("http://a")),
+  Optional.empty(),
+  Optional.empty()
+));
+
+// false
+System.out.println(policy.allowsExternalScript(
+  Optional.empty(),
+  Optional.empty(),
+  Optional.empty(),
+  Optional.empty(),
+  Optional.empty()
+));
+```
+
+Note that these methods were correct according to current draft of the CSP specification when this library was written, but no browser implements precisely the current draft, and changes to the specification may also invalidate assumptions this library makes. There is no alternative to testing on the browsers you care about.
 
-Notice.getAllErrors(notices).get(0).show(); 
-// 1:1: Unrecognised directive-name: "image-src".
+Because the `Policy` objects are rich structures, you can also ask about the presence or absence of specific directives or expressions:
 
-Notice.getAllWarnings(notices).get(0).show();
-// 1:25: This host name is unusual, and likely meant to be a keyword that is missing the required quotes: 'none'.
+```java
+Policy policy = Policy.parseSerializedCSP("script-src 'strict-dynamic'", Policy.PolicyErrorConsumer.ignored);
 
-Notice.getAllInfos(notices).get(0).show(); 
-// 1:31: A draft of the next version of CSP deprecates report-uri in favour of a new report-to directive.
+// true
```

Comment:
I would prefer the comment here to translate this into the question that's being asked. Something like "Does this policy have a `script-src` directive, and if so, does it contain `'strict-dynamic'`?".